lib/rules/base.js

```diff
@@ -75,6 +76,54 @@ module.exports = class BaseRule {
     return name === this.ruleName || DEPRECATED_RULES.get(name) === this.ruleName;
   }
 
+  _determineConfigForSeverity(config) {
+    switch (config) {
+      case 'off':
+        return false;
+      case 'warn':
+        this.severity = 1;
+        return true;
+      case 'error':
+        this.severity = 2;
+        return true;
+    }
+  }
+
+  /**
+   * Sets the severity of the rule based on user configuration
+   * and returns the config based on the severity OR as defined by the user.
+   * Eg:
+   * {'no-implicit-this': 'warn'}
+   * This would set the severity to 1 and return `true`.
+   * { 'no-implicit-this': { "severity": "warn", "config": "allow": [ "fooData" ] } }
+   * This would set the severity of the rule to 1 and return the user defined config back.
+   * { 'no-implicit-this': { "severity": "warn"} }
+   * This would set the severity of the rule to 1 and return true.
+   * @param {*} configData
+   */
+  processConfigForSeverity(configData) {
+    let severityConfig;
+    // In case of {'no-implicit-this': 'off|warn|error'}
+    if (typeof configData === 'string') {
+      severityConfig = this._determineConfigForSeverity(configData);
+      return severityConfig !== undefined ? severityConfig : configData;
+    }
+
+    // In case of object based config with or without severity information.
+    if (typeof configData === 'object') {
```

Comment:
I don't know exactly how to make this change compatibly. It is already possible to have a rule that accepts a string, object, array, set, etc (anything really)...